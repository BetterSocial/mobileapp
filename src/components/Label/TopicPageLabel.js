import * as React from 'react';
import {StyleSheet, Text, View} from 'react-native';

import {COLORS} from '../../utils/theme';
import {fonts, normalize, normalizeFontSize} from '../../utils/fonts';

const TopicPageLabel = ({label}) => (
  <>
    <View style={styles.container}>
      <Text style={styles.domainText} numberOfLines={1} ellipsizeMode="tail">
        {label}
      </Text>
    </View>
    <View style={styles.divider} />
  </>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    marginStart: 20,
    alignSelf: 'flex-start',
    backgroundColor: COLORS.almostBlack,
    height: normalize(29)
  },
  domainText: {
    fontSize: normalizeFontSize(18),
    fontFamily: fonts.inter[600],
    lineHeight: normalize(29),
    fontWeight: 'bold',
    textAlign: 'left',
    color: COLORS.white2
  },
  divider: {
    height: 1,
    width: '100%',
    backgroundColor: COLORS.gray400
  }
});

export default TopicPageLabel;
