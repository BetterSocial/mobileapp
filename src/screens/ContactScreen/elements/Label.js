import * as React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {COLORS} from '../../../utils/theme';

const Label = ({
  label,
  containerBgColor = COLORS.gray100,
  titleHeaderColor = COLORS.almostBlackSmoke,
  textColor = COLORS.white
}) => (
  <View style={styles.headerList(containerBgColor)}>
    <Text style={styles.titleHeader(titleHeaderColor)}>
      <Text style={styles.textBold(textColor)}>{label}</Text>
    </Text>
  </View>
);

export default Label;

const styles = StyleSheet.create({
  headerList: (containerBgColor) => ({
    height: 40,
    paddingLeft: 22,
    paddingRight: 22,
    backgroundColor: containerBgColor,
    flexDirection: 'column',
    justifyContent: 'center'
  }),
  titleHeader: (titleHeaderColor) => ({
    fontFamily: 'Poppins',
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 12,
    lineHeight: 18,
    color: titleHeaderColor
  }),
  textBold: (textColor) => ({
    fontFamily: 'Poppins',
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: 12,
    lineHeight: 18,
    color: textColor,
    textTransform: 'capitalize'
  })
});
