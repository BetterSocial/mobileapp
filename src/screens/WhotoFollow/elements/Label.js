import * as React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {COLORS} from '../../../utils/theme';
import dimen from '../../../utils/dimen';
import {normalizeFontSize} from '../../../utils/fonts';

const Label = ({label, isOriginalText}) => (
  <View style={styles.headerList}>
    {isOriginalText ? (
      <Text style={styles.titleHeader}>{label}</Text>
    ) : (
      <Text style={styles.titleHeader}>
        People in <Text style={styles.textBold}>{label}</Text> follow...
      </Text>
    )}
  </View>
);

export default Label;

const styles = StyleSheet.create({
  headerList: {
    height: dimen.normalizeDimen(40),
    paddingLeft: dimen.normalizeDimen(22),
    paddingRight: dimen.normalizeDimen(22),
    backgroundColor: COLORS.gray100,
    flexDirection: 'column',
    justifyContent: 'center'
    // marginBottom: 12,
    // marginTop: 12,
  },
  titleHeader: {
    fontFamily: 'Poppins',
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: normalizeFontSize(12),
    lineHeight: normalizeFontSize(18),
    color: COLORS.gray500
  },
  textBold: {
    fontFamily: 'Poppins',
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: normalizeFontSize(12),
    lineHeight: normalizeFontSize(18),
    color: COLORS.black000
  }
});
