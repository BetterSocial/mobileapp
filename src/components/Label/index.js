import TopicPageLabel from "./TopicPageLabel"

export default {
    TopicPageLabel
}