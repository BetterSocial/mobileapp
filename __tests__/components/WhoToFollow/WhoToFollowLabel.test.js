import * as React from 'react'
import renderer from 'react-test-renderer'

import Label from '../../../src/screens/WhotoFollow/elements/Label'

describe('Label render correctly', () => {
    it('Match snapshot', () => {
        const tree = renderer.create(<Label label='#Veterans' />).toJSON()
        expect(tree).toMatchSnapshot()    
    })
})