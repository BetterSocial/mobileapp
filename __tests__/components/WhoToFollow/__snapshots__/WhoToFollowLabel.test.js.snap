// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label render correctly Match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F2F2",
      "flexDirection": "column",
      "height": 40,
      "justifyContent": "center",
      "paddingLeft": 22,
      "paddingRight": 22,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4F4F4F",
        "fontFamily": "Poppins",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "lineHeight": 18,
      }
    }
  >
    People in 
    <Text
      style={
        Object {
          "color": "#4F4F4F",
          "fontFamily": "Poppins",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "bold",
          "lineHeight": 18,
        }
      }
    >
      #Veterans
    </Text>
     follow...
  </Text>
</View>
`;
